libmachine/provision/utils.go

```diff
@@ -172,10 +174,17 @@ func ConfigureAuth(p Provisioner) error {
 		return err
 	}
 
-	// TODO: Do not hardcode daemon port, ask the driver
-	if err := utils.WaitForDocker(ip, dockerPort); err != nil {
-		return err
-	}
+	waitForDocker(p, dockerPort)
 
 	return nil
 }
+
+func waitForDocker(p Provisioner, dockerPort int) {
+	utils.WaitForSpecific(func() bool {
+		// HACK: Check netstat's output to see if anyone's listening on the Docker API port.
+		if _, err := p.SSHCommand(fmt.Sprintf("netstat -a | grep %d", dockerPort)); err != nil {
```

Comment:
(`grep` is going to be much more wildly varying than `netstat`, in my experience, but it's worth testing a few different distros like `busybox`, `debian`, and something RPM-based like `fedora` to verify `netstat` output is reasonably sane)